code/game/objects/items/dehy_carp.dm

```diff
@@ -6,22 +6,21 @@
 // Child of carpplushie because this should do everything the toy does and more
 /obj/item/toy/carpplushie/dehy_carp
 	var/mob/owner = null	// Carp doesn't attack owner, set when using in hand
-	var/owned = 1	// Boolean, no owner to begin with
+	var/owned = 0	// Boolean, no owner to begin with
 
 // Attack self
 /obj/item/toy/carpplushie/dehy_carp/attack_self(mob/user)
 	src.add_fingerprint(user)	// Anyone can add their fingerprints to it with this
-	if(owned)
-		user << "<span class='notice'>[src] stares up at you with friendly eyes.</span>"
+	if(!owned)
+		user << "<span class='notice'>You pet [src]. You swear it looks up at you.</span>"
 		owner = user
-		owned = 0
+		owned = 1
 	return ..()
 
 
 /obj/item/toy/carpplushie/dehy_carp/afterattack(obj/O, mob/user,proximity)
 	if(!proximity) return
 	if(istype(O,/obj/structure/sink))
```

Comment:
I used monkeycubes (amongst other things) as a basis since I hadn't written any other DM code before and I didn't understand what IRC or coderbus was.

So yes, they should have the same parent really because they do the same thing, if you want to go that far I think you'd need to check Chemistry-Reagents.dm since there's two separate reactions for the carp plushie and the monkey cube but I didn't know then how I could have combined the two.

#8777 was the original PR.
#9150 fixed a runtime and made the kerp work for nuclear operatives.